ClavaLaraApi/src-lara-clava/clava/clava/pass/SingleReturnFunction.js

```diff
@@ -0,0 +1,41 @@
+laraImport("lara.pass.Pass");
+laraImport("weaver.Query");
+
+class SingleReturnFunction extends Pass {
+  constructor() {
+    super("SingleReturnFunction");
+  }
+
+  _apply_impl($jp) {
+    if ($jp.joinpointType !== "function" || !$jp.isImplementation) {
+      return;
+    }
+    const $body = $jp.body;
+    const $returnStmts = Array.from(Query.searchFrom($body, "returnStmt"));
+    if ($returnStmts.length <= 1) {
```

Comment:
I thought it would not be allowed for non-void functions. Will change it to take it into account.

For a given type, is the default return value the same as the value of an uninitialised variable?

Reply:
The implicit return value of a C function seems to also be undefined (except for the main() function):

https://stackoverflow.com/a/10079465